models/events.go

```diff
@@ -82,6 +83,19 @@ func (c *CadenceEvents) Transactions() ([]Transaction, []*StorageReceipt, error)
 		}
 	}
 
```

Comment:
Why not just change the above for loop:

```go
        cumulativeGasUsed := uint64(0)
        
        for _, e := range c.events.Events {
		if isTransactionExecutedEvent(e.Value) {
			tx, receipt, err := decodeTransactionEvent(e.Value)
			if err != nil {
				return nil, nil, err
			}
			
			// dynamically calculated values
			cumulativeGasUsed += receipt.GsUsed
			receipt.CumulativeGasUsed = cumulativeGasUsed
			
			txs = append(txs, tx)
			rcps = append(rcps, receipt)
		}
	}
```

Reply:
also can you please add a simple unit test for this, there should already be an existing test for `Transactions()` you just need to check cumulative.

Reply:
Updated the loop in https://github.com/onflow/flow-evm-gateway/pull/358/commits/b056d0183c50896501adb1f54d98a4105aaf3012